src/Mutation.jl

```diff
@@ -1,7 +1,58 @@
 module Mutation
 
 using ..Types
+using Distributions
 
-# TODO
+"""
+    Defines the mutation of genes of real values (currently only Float64 and Integer types (including Bool))
+    Mutation is applied for each gene on its own with probability `mutation_rate` vith values from the uniform
+    distribution in the interval 'mutation_interval'.
+"""
+struct RealGeneMutation{T<:Union{Float64, Integer}}
+    mutation_rate::Float64
+    mutation_interval::Tuple{T, T}
 
+    function RealGeneMutation(mutation_rate::Float64, mutation_interval::Tuple{T, T}) where T<:Real
+        if mutation_rate < 0 || mutation_rate > 1
+            throw(ArgumentError("Mutation rate must be between 0 and 1"))
+        end
+        if mutation_interval[1] > mutation_interval[2]
+            throw(ArgumentError("Mutation interval must be in the form (min, max)"))
+        end
+        if mutation_interval[1] == mutation_interval[2]
+            throw(ArgumentError("Mutation interval must have a range"))
+        end
+        new{eltype(mutation_interval)}(mutation_rate, mutation_interval)
+    end
+
+end
+
+"""
+    Mutates the genes with a probability of c.mutation_rate and values in the interval c.mutation_interval
+"""
+function (c::RealGeneMutation)(genes::Matrix{T})::Matrix{T} where T<:Float64
+    
+    mask = rand((0,1), size(genes)) .< c.mutation_rate
```

Comment:
Use also Uniform(0, 1) to create the mask. `rand((0,1), size(genes))` does only produce 0s and 1s.